src/modules/ekf2/EKF/aid_sources/external_vision/ev_vel_control.cpp

```diff
@@ -256,92 +144,64 @@ void Ekf::controlEvVelFusion(const imuSample &imu_sample, const extVisionSample
 	}
 }
 
-bool Ekf::fuseEvVelocity(estimator_aid_source3d_s &aid_src, const extVisionSample &ev_sample)
+void Ekf::stopEvVelFusion()
 {
-	if (ev_sample.vel_frame == VelocityFrame::BODY_FRAME_FRD) {
-
-		VectorState H;
-		estimator_aid_source1d_s current_aid_src;
-		const auto state_vector = _state.vector();
-
-		for (uint8_t index = 0; index <= 2; index++) {
-			current_aid_src.timestamp_sample = aid_src.timestamp_sample;
+	if (_control_status.flags.ev_vel) {
+		_control_status.flags.ev_vel = false;
+	}
+}
 
-			if (index == 0) {
-				sym::ComputeEvBodyVelHx(state_vector, &H);
+void Ekf::fuseLocalFrameVelocity(estimator_aid_source3d_s &aid_src, const uint64_t &timestamp,
+				 const Vector3f &measurement, const Vector3f &measurement_var, const float &innovation_gate)
+{
+	updateAidSourceStatus(aid_src,
+			      timestamp,				// sample timestamp
+			      measurement,				// observation
+			      measurement_var,				// observation variance
+			      _state.vel - measurement,			// innovation
+			      getVelocityVariance() + measurement_var,	// innovation variance
+			      innovation_gate);				// innovation gate
+	fuseVelocity(aid_src);
+}
 
-			} else if (index == 1) {
-				sym::ComputeEvBodyVelHy(state_vector, &H);
+void Ekf::fuseBodyFrameVelocity(estimator_aid_source3d_s &aid_src, const uint64_t &timestamp,
+				const Vector3f &measurement, const Vector3f &measurement_var, const float &innovation_gate)
+{
+	const auto state_vector = _state.vector();
 
-			} else {
-				sym::ComputeEvBodyVelHz(state_vector, &H);
-			}
+	matrix::Vector<VectorState, 3> H;
```

Comment:
You could use an array of VectorSate instead of a vector of VectorState since you don't need any special operators from the Vector class